selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/executor/hotupdate/HotUpdateUtil.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core.executor.hotupdate;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.eclipse.che.selenium.core.client.TestUserPreferencesServiceClient;
+import org.eclipse.che.selenium.core.executor.OpenShiftCliCommandExecutor;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.core.webdriver.WebDriverWaitFactory;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+
+@Singleton
+public class HotUpdateUtil {
+  private static final int TIMEOUT_FOR_FINISH_UPDATE_IN_SECONDS = 600;
+  private static final String UPDATE_COMMAND = "rollout latest che";
+  private static final String PODS_LIST_COMMAND = "get pods | awk 'NR > 1 {print $1}'";
+  private static final String COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE =
+      "get dc | grep che | awk '{print $2}'";
+
+  private final OpenShiftCliCommandExecutor openShiftCliCommandExecutor;
+  private final WebDriverWaitFactory webDriverWaitFactory;
+  private final TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+
+  @Inject
+  HotUpdateUtil(
```

Comment:
It's mostly about _Rolling_ update, so the name `HotRollingUpdateUtil` better describes class propose, IMHO

Reply:
It is also used in the recreate test, and all methods do not have a dependency to update strategy.